api/dps/index_test.go

```diff
@@ -0,0 +1,456 @@
+						assert.Equal(t, path2[:], in.Paths[1])
+					}
+					assert.Equal(t, in.Height, testHeight)
+
+					return &GetRegistersResponse{
+						Height: testHeight,
+						Paths:  convert.PathsToBytes(testPaths),
+						Values: convert.ValuesToBytes(testValues),
+					}, nil
+				},
+			},
+		}
+
+		got, err := index.Registers(testHeight, testPaths)
+
+		if assert.NoError(t, err) {
+			assert.Equal(t, testValues, got)
+		}
+	})
+
+	t.Run("handles indexing failures", func(t *testing.T) {
+		index := Index{
+			client: &apiMock{
+				GetRegistersFunc: func(_ context.Context, in *GetRegistersRequest, _ ...grpc.CallOption) (*GetRegistersResponse, error) {
+					if assert.Len(t, in.Paths, 2) {
+						assert.Equal(t, path1[:], in.Paths[0])
+						assert.Equal(t, path2[:], in.Paths[1])
+					}
+					assert.Equal(t, in.Height, testHeight)
+
+					return nil, mocks.DummyError
+				},
+			},
+		}
+
+		_, err := index.Registers(testHeight, testPaths)
+
+		assert.Error(t, err)
+	})
+}
+
+func TestIndex_Height(t *testing.T) {
+	testHeight := uint64(42)
+	testBlockID, err := flow.HexStringToIdentifier("98827808c61af6b29c7f16071e69a9bbfba40d0f96b572ce23994b3aa605c7c2")
+	require.NoError(t, err)
+
+	t.Run("nominal case", func(t *testing.T) {
+		index := Index{
+			client: &apiMock{
+				GetHeightFunc: func(_ context.Context, in *GetHeightRequest, _ ...grpc.CallOption) (*GetHeightResponse, error) {
+					assert.Equal(t, testBlockID[:], in.BlockID)
+
+					return &GetHeightResponse{
+						BlockID: testBlockID[:],
+						Height:  testHeight,
+					}, nil
+				},
+			},
+		}
+
+		got, err := index.Height(testBlockID)
+
+		if assert.NoError(t, err) {
+			assert.Equal(t, testHeight, got)
+		}
+	})
+
+	t.Run("handles indexing failures", func(t *testing.T) {
+		index := Index{
+			client: &apiMock{
+				GetHeightFunc: func(_ context.Context, in *GetHeightRequest, _ ...grpc.CallOption) (*GetHeightResponse, error) {
+					assert.Equal(t, testBlockID[:], in.BlockID)
+
+					return nil, mocks.DummyError
+				},
+			},
+		}
+
+		_, err := index.Height(testBlockID)
+
+		assert.Error(t, err)
+	})
+}
+
+type apiMock struct {
+	GetFirstFunc     func(ctx context.Context, in *GetFirstRequest, opts ...grpc.CallOption) (*GetFirstResponse, error)
+	GetLastFunc      func(ctx context.Context, in *GetLastRequest, opts ...grpc.CallOption) (*GetLastResponse, error)
+	GetHeaderFunc    func(ctx context.Context, in *GetHeaderRequest, opts ...grpc.CallOption) (*GetHeaderResponse, error)
+	GetCommitFunc    func(ctx context.Context, in *GetCommitRequest, opts ...grpc.CallOption) (*GetCommitResponse, error)
+	GetEventsFunc    func(ctx context.Context, in *GetEventsRequest, opts ...grpc.CallOption) (*GetEventsResponse, error)
+	GetRegistersFunc func(ctx context.Context, in *GetRegistersRequest, opts ...grpc.CallOption) (*GetRegistersResponse, error)
+	GetHeightFunc    func(ctx context.Context, in *GetHeightRequest, opts ...grpc.CallOption) (*GetHeightResponse, error)
+}
```

Comment:
Might want to move this into the testing package. I imagine we can re-use it for some other tests?

Reply:
IMO it's better to keep them here until the day where we need to re-use it. Otherwise I'm afraid we'll bloat our testing package and might even end up with unused mocks in there. WDYT?